src/TPPCore.Client.Common/CommonClient.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Newtonsoft.Json.Linq;
+using System;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace TPPCore.Client.Common
+{
+    public class CommonClient
+    {
+        public static async Task<string> GetAsync(Uri requestUri, RestfulClient HttpClient)
+        {
+            Task<string> task = HttpClient.GetStringAsync(requestUri);
+            string msg = await task;
+            return msg;
+        }
+
+        public static async Task PostAsync(Uri requestUri, string message, RestfulClient HttpClient)
+        {
+            HttpResponseMessage msg = await HttpClient.PostAsync(requestUri, new StringContent(message));
+            if (!msg.IsSuccessStatusCode)
+                throw new HttpRequestException();
+        }
+
+        public static string Escape(string toEscape)
+        {
+            return Regex.Replace(toEscape, @"[^\w%]", m => "%" + ((int)m.Value[0]).ToString("X2"));
```

Comment:
Why not use `Uri.EscapeDataString` instead? It handles recognizing reserved characters and also Unicode strings.